lib/defaults.js

```diff
@@ -15,11 +15,11 @@ function setContentTypeIfUnset(headers, value) {
 
 function getDefaultAdapter() {
   var adapter;
-  if (typeof XMLHttpRequest !== 'undefined') {
```

Comment:
I'd be wary of global window variables

Reply:
I admit that you are right, because we use the XMLHttpRequest object in axios, so judging whether it exists or not is really the best way to judge the browser environment; 
So I will revert the if condition(`typeof XMLHttpRequest !== 'undefined'`) which judging whether it is browsing environment. if not, then we thought it is the nodejs environment by default.

Reply:
also please add tests 🙏

Reply:
It will take me some time to add the tests for this paticular scenario...wait for me :joy: